modules/mssql/mssql_test.go

```diff
@@ -0,0 +1,210 @@
+		}
+	})
+
+	// perform assertions
+	connectionString, err := container.ConnectionString(ctx)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db, err := sql.Open("sqlserver", connectionString)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	if err = db.Ping(); err != nil {
+		t.Errorf("error pinging db: %+v\n", err)
+	}
+
+	var tableName string
+	err = db.QueryRow("SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = N'a_table'").Scan(&tableName)
+
+	if err == sql.ErrNoRows {
+		_, err = db.Exec("CREATE TABLE a_table ( " +
+			" [col_1] NVARCHAR(128) NOT NULL, " +
+			" [col_2] NVARCHAR(128) NOT NULL, " +
+			" PRIMARY KEY ([col_1], [col_2]) " +
+			")")
+		if err != nil {
+			t.Errorf("error creating table: %+v\n", err)
+		}
+	} else if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestMSSQLServerWithConnectionStringParameters(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx,
+		WithAcceptEULA("Y"),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	// perform assertions
+	connectionString, err := container.ConnectionString(ctx, "encrypt=false", "TrustServerCertificate=true")
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db, err := sql.Open("sqlserver", connectionString)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	if err = db.Ping(); err != nil {
+		t.Errorf("error pinging db: %+v\n", err)
+	}
+
+	var tableName string
+	err = db.QueryRow("SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = N'a_table'").Scan(&tableName)
+
+	if err == sql.ErrNoRows {
+		_, err = db.Exec("CREATE TABLE a_table ( " +
+			" [col_1] NVARCHAR(128) NOT NULL, " +
+			" [col_2] NVARCHAR(128) NOT NULL, " +
+			" PRIMARY KEY ([col_1], [col_2]) " +
+			")")
+		if err != nil {
+			t.Errorf("error creating table: %+v\n", err)
+		}
+	} else if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestMSSQLServerWithCustomStrongPassword(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx,
+		WithAcceptEULA("Y"),
+		WithPassword("Strong@Passw0rd"),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	// perform assertions
+	connectionString, err := container.ConnectionString(ctx)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	db, err := sql.Open("sqlserver", connectionString)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	if err = db.Ping(); err != nil {
+		t.Errorf("error pinging db: %+v\n", err)
+	}
+}
+
+// tests that a weak password is not accepted by the container due to Microsoft's password strength policy
+func TestMSSQLServerWithInvalidPassword(t *testing.T) {
```

Comment:
microsoft's docker service will handle the password validation, since they have a minimum criteria for their passwords. This test confirms this.
[source](https://learn.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver16)